### PR TITLE
Feature: External Markdown Descriptions

### PR DESCRIPTION
**Pull request by jlacivita**
- Added a descriptions module to load .md files stored in /src/descriptions/
- Updated json-schema module to use descriptions module to inject md
- Using these new features in build:openrpc and build:docs

### CODE REVIEW

File: util/shared/descriptions.mjs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+// A through stream that expects a stream of filepaths, reads the contents
+// of any .json files found, and converts them to POJOs
+// DOES NOT DEAL WITH ERRORS
+const loadMarkdownContent = fileStream => fileStream
+    .filter(filepath => path.extname(filepath) === '.md')
+    .flatMap(filepath => {
+        return fsReadFile(filepath)
+            .map(bufferToString)
+            .tap(data => descriptions[filepath.split(pathDelimiter)[1]] = data)
```
Comment (cletustboone) [inline util/shared/descriptions.mjs:33]: Not asking for a change here, but as info, Node has a helpful module for doing this kind of stuff in a platform-agnostic way.

https://nodejs.org/api/path.html

File: util/shared/json-schema.mjs [hunk 2]
```diff
@@ -36,6 +41,70 @@ const getAllSchemas = _ => {
...
+    const field = path.pop()
+
+    // reassign value
+    const node = getPathOr(null, path, obj)
+    node[field] = md
+  })
+
+  return obj
+
+/*
```
Comment (cletustboone) [inline util/shared/json-schema.mjs:73]: Is there a reason to keep this lying around?

File: util/shared/descriptions.mjs (hunk 1 above)
Comment (jlacivita) [reply util/shared/descriptions.mjs:33]: Good to know, thanks!

File: util/shared/json-schema.mjs (hunk 2 above)
Comment (jlacivita) [reply util/shared/json-schema.mjs:73]: Well, it's used in a stream to add all of the markdown content to the obj passed in. if we don't return it, it wouldn't be very streamy. If we take this out, we'd have to change the `.map()` to a `.tap()` in the `getSchemaContent` method.

At the moment it's the final step in the stream, but other steps could be added in the future.

Comment (jlacivita) [reply util/shared/json-schema.mjs:73]: I suppose you could make the case that it should make a copy of the object and do its operations.

File: util/shared/modules.mjs [hunk 3]
```diff
@@ -204,11 +205,11 @@ const getAllModules = _ => {
 // A through stream that expects a stream of filepaths, reads the contents
 // of any .json files found, and converts them to POJOs
 // DOES NOT DEAL WITH ERRORS
-const getModuleContent = fileStream => fileStream
-    .filter(filepath => path.extname(filepath) === '.json')
-    .flatMap(fsReadFile)
-    .map(bufferToString)
-    .map(JSON.parse)
+const getModuleContent = getSchemaContent //fileStream => fileStream
```
[the remark this answers is not in the document]
Comment (jlacivita) [reply util/shared/modules.mjs:208]: good catch, thx

**Review by cletustboone — APPROVED** [review]
The changes make sense and seem fine to me. I'd get rid of some of the commented out code though.

File: util/shared/json-schema.mjs (hunk 2 above)
Comment (jlacivita) [reply util/shared/json-schema.mjs]: Oh... 😳

Uhm... I wanted "Jeremy was here" in the commit history?

Fixed.

**Review by cletustboone — APPROVED** [review]
👍🏻

**Pull request merged**